---work/unpatched/net/minecraft/server/PlayerList.java
+++patched/net/minecraft/server/PlayerList.java
@@ -1,8 +1,10 @@
 package net.minecraft.server;
 
 import co.aikar.timings.MinecraftTimings;
+import lombok.*;
+
 import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
 import com.mojang.authlib.GameProfile;
 import io.netty.buffer.Unpooled;
@@ -39,10 +39,11 @@
 import org.bukkit.event.player.PlayerLoginEvent;
 import org.bukkit.event.player.PlayerQuitEvent;
 import org.bukkit.event.player.PlayerRespawnEvent;
 import org.bukkit.event.player.PlayerTeleportEvent.TeleportCause;
 import org.bukkit.util.Vector;
+import org.fountainmc.taco.TacoFountainServer;
 import org.spigotmc.event.player.PlayerSpawnLocationEvent;
 // CraftBukkit end
 
 public abstract class PlayerList {
 
@@ -66,20 +66,28 @@
     private int r;
     private EnumGamemode s;
     private boolean t;
     private int u;
 
+    // Fountain start
+    @Getter
+    private final TacoFountainServer fountainServer;
+    // Fountain end
     // CraftBukkit start
     private CraftServer cserver;
     private final Map<String,EntityPlayer> playersByName = new org.spigotmc.CaseInsensitiveMap<EntityPlayer>();
     @Nullable String collideRuleTeamName; // Paper - Team name used for collideRule
 
     public PlayerList(MinecraftServer minecraftserver) {
         this.cserver = minecraftserver.server = new CraftServer(minecraftserver, this);
         minecraftserver.console = org.bukkit.craftbukkit.command.ColouredConsoleSender.getInstance();
         minecraftserver.reader.addCompleter(new org.bukkit.craftbukkit.command.ConsoleCommandCompleter(minecraftserver.server));
         // CraftBukkit end
+        // Fountain start
+        this.fountainServer = new TacoFountainServer((DedicatedServer) minecraftserver, this);
+        // Fountain end
 
         this.k = new GameProfileBanList(PlayerList.a);
         this.l = new IpBanList(PlayerList.b);
         this.operators = new OpList(PlayerList.c);
         this.whitelist = new WhiteList(PlayerList.d);
